### PR TITLE
[AIT-1141] fix(uts/objects): spec-side corrections from the LiveObjects deviations audit

### PR DESCRIPTION
**Pull request by sacOO7**
- Fixes https://ably.atlassian.net/browse/AIT-1141

## Summary

Spec-side corrections produced by a comprehensive audit that reconciled ably-js LiveObjects against the objects spec (`objects-features.md` + `uts/objects`), treating the reference implementation as the source of truth for behaviours that have already shipped. Every UTS unit test body was cross-checked against its derived implementation; the changes below fix spec tests that were unrunnable for a compliant SDK, assertions that contradicted the shipped public API, and pseudocode gaps that forced SDKs to document avoidable "deviations".

Companion implementation PR: ably/ably-js#2263 (source fixes + derived tests, kept in lock-step with the test IDs and bodies changed here).

### 1. Unsolicited-server-DETACHED tests were unrunnable — switched to client-side detach

`RTO25b/access-throws-detached-0`, `RTO26b/write-throws-detached-0` and the RTO20e1 test injected an unsolicited server `DETACHED` and then asserted `AWAIT_STATE channel.state == DETACHED`. A compliant SDK immediately re-attaches on an unsolicited DETACHED (RTL13a), so the DETACHED state is transient, the `AWAIT_STATE` is a race, and — since every mock answers the re-ATTACH with ATTACHED — the precondition under test evaporates. This is the same reasoning that moved the proxy tier to an ERROR-based sequence in #501, and the spec's own RTPO19b test already used the correct pattern.

All three now reach DETACHED via a client-side `AWAIT channel.detach()` (solicited detach does not trigger RTL13a), and the shared helper mocks (`setup_synced_channel`, `setup_synced_channel_no_ack`) gain the DETACH → DETACHED responder the tests rely on.

### 2. RTO20e1 now covers both reachable states

The single DETACHED-injection test is replaced by a pair:

- `objects/unit/RTO20e1/fails-on-channel-detached-0` — client-side detach while the operation waits for SYNCED.
- `objects/unit/RTO20e1/fails-on-channel-failed-0` (new) — channel ERROR → FAILED, the same sequence as the proxy-tier test.

SUSPENDED is a connection-level state and is noted as out of scope for a channel-level unit mock.

### 3. RTO15 test asserts observable behaviour only

The test obtained `result = AWAIT channel.object.publish([...])` and asserted `result.serials` — but RTO15 defines an **internal** function, and the realtime `channel.object` exposes no public `publish`. The test now drives the publish through a public mutation and asserts the outbound OBJECT ProtocolMessage shape; the `PublishResult.serials` contract is observed through its only consumer, the neighbouring RTO20 apply-on-ACK tests. The `RTO15h` clause itself is unchanged.

### 4. Public map semantics asserted as `"lww"`

The shipped public type is lowercase (`ObjectsMapSemantics = 'lww' | 'unknown'` in ably-js); asserting `"LWW"` could never match the public API. Five assertions in `public_object_message.md` updated. *Reviewer note: please confirm ably-java's public rendering agrees — if it shipped uppercase, this becomes a cross-SDK reconciliation instead.*

### 5. Pseudocode conventions and language-applicability notes

- `uts/README.md` gains a conventions section: `== null` means the language-appropriate absent value (`undefined` in JS); member access written as a call (`instance.id()`) is satisfied by a property where the features spec defines one (RTINS3); inputs that cannot be constructed in a language (non-string map keys in JS, `null` where it is indistinguishable from "omitted") make that test/row not applicable — sanctioned omissions rather than coverage gaps.
- The `RTLC12e1` invalid-amounts table's `null` row and the `RTLMV4b` key-type test carry explicit applicability notes referencing those conventions.

This lets SDKs retire boilerplate "deviation" records for what are really language-mapping conventions.

### 6. Small fixes

- `RTO24c1` subscribe call flipped to listener-first (`subscribe(listener, { depth: 1 })`), matching every other with-options call after the argument-order convergence.
- `objects_faults.md` unit-tier cross-reference updated to `fails-on-channel-failed-0`, the test whose ERROR-based sequence it describes.

## Verification

The derived ably-js tests for every changed body pass spec-verbatim (full UTS unit suite: 1351 passing) — see ably/ably-js#2263 for the implementation side, including the two runtime bugs the same audit surfaced (sub-16-character object-id nonces, and one malformed inbound operation discarding its siblings).

## Follow-ups (out of scope here)

- Nil operation payloads (`COUNTER_INC` without `counterInc`, `MAP_SET` without `mapSet`, `MAP_REMOVE` without `mapRemove`) are spec-silent; ably-js now warns-and-skips them consistently with RTLC7d3/RTLM15d4. Proposing explicit sub-clauses is left as a follow-up.
- `uts-liveobjects` predates `main`'s `OD2g`/`OD3g` (`json` ObjectData) clauses — a merge from `main` is recommended before further UTS reconciliation.

### CODE REVIEW

File: uts/objects/unit/realtime_object.md [hunk 1]
```diff
@@ -240,7 +239,6 @@ ASSERT captured_messages.length == 1
 ASSERT captured_messages[0].action == OBJECT
 ASSERT captured_messages[0].channel == "test"
 ASSERT captured_messages[0].state.length == 1
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply uts/objects/unit/realtime_object.md:241]: Agreed — dropping the `result.serials` assertion left this test verifying only that *some* state entry exists, which under-covers RTO15e3. Fixed in 9641a81e: the assertions now pin the state entry to the encoded operation for the driven mutation (`operation.action == COUNTER_INC`, `objectId == "counter:score@1000"`, `counterInc.number == 5`). The derived ably-js test mirrors this in ably/ably-js@2cfe8eff.

File: uts/objects/unit/realtime_object.md [hunk 2]
```diff
@@ -420,9 +423,51 @@ mock_ws.send_to_client(ProtocolMessage(
 
 inc_future = root.get("score").increment(10)
 
+# The publish and its ACK complete against the mock; publishAndApply parks in the
+# RTO20e wait for SYNCED. A client-side detach then moves the channel to DETACHED.
+AWAIT channel.detach()
+
+AWAIT inc_future FAILS WITH error
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply uts/objects/unit/realtime_object.md:437]: Agreed — and the false-positive is real: an SDK that *rejects* early with 92008 (without implementing the RTO20e wait at all) would have passed, since only an early-resolve was caught by the `FAILS WITH` step. Fixed in 9641a81e with `ASSERT inc_future IS NOT complete` before the detach, matching the RTO20e waits-for-synced pattern. Mirrored in the derived ably-js test (ably/ably-js@2cfe8eff) using the same settled-flag rendering as the existing RTO20e test.

File: uts/objects/unit/realtime_object.md [hunk 3]
```diff
@@ -420,9 +423,51 @@ mock_ws.send_to_client(ProtocolMessage(
...
+inc_future = root.get("score").increment(10)
+
+# The publish and its ACK complete against the mock; publishAndApply parks in the
+# RTO20e wait for SYNCED. Then the channel ERROR moves the channel to FAILED.
+mock_ws.send_to_client(ProtocolMessage(
+  action: ERROR, channel: "test",
+  error: { code: 90000, statusCode: 400, message: "Channel failed" }
 ))
 
 AWAIT inc_future FAILS WITH error
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply uts/objects/unit/realtime_object.md:483]: Fixed in 9641a81e — same treatment as the DETACHED variant: `ASSERT inc_future IS NOT complete` before the ERROR injection pins the parked-in-wait precondition. Mirrored in the derived ably-js test (ably/ably-js@2cfe8eff).

File: uts/objects/unit/public_object_message.md [hunk 4]
```diff
@@ -371,7 +371,7 @@ Tests that when the source ObjectOperation has a `mapCreate` field, the PublicAP
 source_operation = {
   action: "MAP_CREATE",
   objectId: "map:new@2000",
-  mapCreate: { semantics: "LWW", entries: { "key1": { data: { string: "val1" } } } }
+  mapCreate: { semantics: "lww", entries: { "key1": { data: { string: "val1" } } } }
```
Comment (ttypic) [inline uts/objects/unit/public_object_message.md]: Why it's in lowercase? Doesn't it simply represent enum in actual code?

Comment (sacOO7) [reply uts/objects/unit/public_object_message.md]: You're right — `semantics` is the `ObjectsMapSemantics` enum, wire-encoded as an integer (OMP2: "values in order from zero: LWW"), and the lowercase form here was ably-js's idiomatic rendering (`ObjectsMapSemantics = 'lww'` string-literal union) leaking into the pseudo-code as if it were the canonical contract.

Reverted to the symbolic `"LWW"` in b3943994, and added an **Enum values** bullet to the `uts/README.md` pseudocode conventions so each SDK's derivation is sanctioned explicitly: the enum member itself in typed SDKs (`MapSemantics.LWW` in ably-java), the string-literal union value in ably-js (`'lww'`). The ably-js derived test keeps asserting its real public API and now cites that convention instead of claiming lowercase is spec-mandated.

**Review by ttypic — APPROVED** [review]
LGTM
